packages/flutter/lib/src/animation/animations.dart

```diff
@@ -261,11 +291,13 @@ class TrainHoppingAnimation extends Animation<double>
     assert(_mode != null);
   }
 
+  /// The animation that is current driving this animation.
   Animation<double> get currentTrain => _currentTrain;
   Animation<double> _currentTrain;
   Animation<double> _nextTrain;
   _TrainHoppingMode _mode;
 
+  /// Called when this animation switches to be driven by a different animation.
   VoidCallback onSwitchedTrain;
 
   AnimationStatus _lastStatus;
```

Comment:
there's a bug in _valueChangeHandler.
The line that says:
```
_nextTrain.addListener(_valueChangeHandler);
```
...should be setting a status listener on the next train, not a value listener, and it should pass in _statusChangeHandler, not _valueChangeHandler.